### PR TITLE
8287917: System.loadLibrary does not work on Big Sur if JDK is built with macOS SDK 10.15 and earlier

### PR DESCRIPTION
**Pull request by satoyoshiki**
Please review this PR.
SDK 10.15 and earlier reports os.version as 10.16 on Big Sur.  This fix will check if dynamic linker support, which is supported from Big Sur, is available or not on the OS even if os.version is reported as 10.16 instead of 11.  The os.version 10.16 doesn't include the update version like y of 10.x.y.  Hence the change only see if it is 10.16 or not.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.java.net/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8287917](https://bugs.openjdk.org/browse/JDK-8287917): System.loadLibrary does not work on Big Sur if JDK is built with macOS SDK 10.15 and earlier

### Reviewers
 * [Mandy Chung](https://openjdk.java.net/census#mchung) (@mlchung - **Reviewer**) ⚠️ Review applies to [a9b95965](https://git.openjdk.org/jdk/pull/9077/files/a9b959659e3a7e16e307145c2e5a0da19cb83d30)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9077/head:pull/9077` \
`$ git checkout pull/9077`

Update a local copy of the PR: \
`$ git checkout pull/9077` \
`$ git pull https://git.openjdk.org/jdk pull/9077/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9077`

View PR using the GUI difftool: \
`$ git pr show -t 9077`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9077.diff">https://git.openjdk.org/jdk/pull/9077.diff</a>

</details>

### CODE REVIEW

File: test/jdk/java/lang/RuntimeTests/loadLibrary/exeLibraryCache/LibraryFromCache.java [hunk 1]
```diff
@@ -42,6 +42,8 @@
 
 public class LibraryFromCache {
     public static void main(String[] args) throws IOException {
+        System.out.println("os.version = " + System.getProperty("os.version"));
```
Comment (mlchung) [inline test/jdk/java/lang/RuntimeTests/loadLibrary/exeLibraryCache/LibraryFromCache.java:45]: The copyright end year needs to be updated.

**Review by mlchung — APPROVED** [review]
Looks good.  Thanks for fixing this.

As a background, Big Sur reports 10.16 as backward compatibility when built with macOS SDK 10.15 and earlier.   Hence this needs to check for os version 10.16 that supports dynamic linker cache.

File: test/jdk/java/lang/RuntimeTests/loadLibrary/exeLibraryCache/LibraryFromCache.java (hunk 1 above)
Comment (satoyoshiki) [reply test/jdk/java/lang/RuntimeTests/loadLibrary/exeLibraryCache/LibraryFromCache.java:45]: Thanks for catching.

**Pull request closed without merging**